core/src/main/java/com/google/zxing/oned/Code128Writer.java

```diff
@@ -179,26 +185,28 @@
             break;
           default:
             // Then handle normal characters otherwise
-            switch (codeSet) {
-              case CODE_CODE_A:
-                patternIndex = contents.charAt(position) - ' ';
-                if (patternIndex < 0) {
-                  // everything below a space character comes behind the underscore in the code patterns table
-                  patternIndex += '`';
-                }
-                break;
-              case CODE_CODE_B:
-                patternIndex = contents.charAt(position) - ' ';
-                break;
-              default:
-                // CODE_CODE_C
-                if (position + 1 == length) {
-                  // this is the last character, but the encoding is C, which always encodes two characers
-                  throw new IllegalArgumentException("Bad number of characters for digit only encoding.");
-                }
-                patternIndex = Integer.parseInt(contents.substring(position, position + 2));
-                position++; // Also incremented below
-                break;
+            if ((codeSet == CODE_CODE_A &&
```

Comment:
OK I see. I'm a little concerned about changing the exisitng code path; adding a second one isn't as big a deal. Are these changes necessary vs just having an alternative impl?